manager/notifier.go

```diff
@@ -544,6 +545,77 @@ func (n *notifier) sendEmailNotification(to string, op notificationOp, a *Alert)
 	return writeEmailBody(wc, *smtpSender, to, status, a, n.alertmanagerURL)
 }
 
+// opGenieMessageCreate is the request for sending a opsGenie notification. We are not specifying all the
+// fields. The Details field is populated with the labels from the alert.
+type opsGenieMessageCreate struct {
+	Message     string            `json:"message"`
+	Description string            `json:"description"`
+	Alias       string            `json:"alias"`
+	Teams       []string          `json:"teams"`
+	Details     map[string]string `json:"details"`
+}
+
+// opsGenieMessageClose closes an open alert in OpsGenie
+type opsGenieMessageClose struct {
+	Alias string `json:alias"`
+}
+
+func (n *notifier) sendOpsGenieNotification(op notificationOp, config *pb.OpsGenieConfig, a *Alert) error {
+	// https://api.slack.com/incoming-webhooks
+	var (
+		incidentKey = a.Fingerprint()
+		buf         = []byte{}
+		err         error
+	)
+	switch op {
+	case notificationOpTrigger:
+		msg := &opsGenieMessageCreate{
+			Message:     a.Summary,
+			Description: a.Description,
+			Alias:       strconv.FormatUint(uint64(incidentKey), 10),
+			Teams:       config.Teams,
+			Details:     a.Labels,
```

Comment:
This is not including the `a.Payload` labels anywhere, which can be very useful. If the `Details` field is the only one that may contain a nested JSON map, how about adding two sub-properties in `Details` which each contain a map for the grouping and payload labels?